src/main/kotlin/no/nav/tilleggsstonader/sak/opplysninger/saksbehandler/SaksbehandlerService.kt

```diff
@@ -0,0 +1,119 @@
+    private val behandlingRepository: BehandlingRepository,
+    private val saksbehandlerClient: SaksbehandlerClient,
+) {
+    /**
+     *  I tilfeller hvor saksbehandler manuelt oppretter en revurdering eller en førstegangsbehandling vil oppgaven
+     *  som returneres fra oppgavesystemet være null. Dette skjer fordi oppgavesystemet bruker litt tid av variabel
+     *  lengde på å opprette den tilhørende behandle-sak-oppgaven til den opprettede behandlingen.
+     *
+     * [no.nav.tilleggsstonader.sak.opplysninger.oppgave.dto.SaksbehandlerRolle.OPPGAVE_FINNES_IKKE_SANNSYNLIGVIS_INNLOGGET_SAKSBEHANDLER]: Dersom null blir returnert og
+     * behandlingen befinner seg i steget REVURDERING_ÅRSAK, VILKÅR eller BEREGNE_YTELSE anser vi det som svært
+     * sannsynlig at det er den innloggede saksbehandleren som er ansvarlig for behandlingen - oppgaven har bare ikke
+     * rukket å bli opprettet enda. I dette tilfellet returnerer vi OPPGAVE_FINNES_IKKE_SANNSYNLIGVIS_INNLOGGET_SAKSBEHANDLER
+     * til frontend.
+     *
+     * [no.nav.tilleggsstonader.sak.opplysninger.oppgave.dto.SaksbehandlerRolle.OPPGAVE_FINNES_IKKE]: Dersom null returneres og behandlingen ikke befinner seg i et av de nevnte
+     * stegene returnerer vi OPPGAVE_FINNES_IKKE til frontend.
+     */
+
+    fun finnSaksbehandler(behandlingId: BehandlingId): SaksbehandlerDto? {
+        val oppgave = hentIkkeFerdigstiltOppgaveForBehandlingGittStegtype(behandlingId)
+        return utledAnsvarligSaksbehandlerForOppgave(behandlingId, oppgave)
+    }
+
+    fun hentIkkeFerdigstiltOppgaveForBehandlingGittStegtype(behandlingId: BehandlingId): OppgaveDomain? {
+        val behandling = behandlingRepository.findByIdOrThrow(behandlingId)
+        val oppgavetyper =
+            when (behandling.steg.tillattFor) {
+                BehandlerRolle.SAKSBEHANDLER -> setOf(Oppgavetype.BehandleSak, Oppgavetype.BehandleUnderkjentVedtak)
+                BehandlerRolle.BESLUTTER -> setOf(Oppgavetype.GodkjenneVedtak)
+                else -> emptySet()
+            }
+        return hentOppgaveSomIkkeErFerdigstilt(behandlingId, oppgavetyper)
+    }
+
+    fun hentOppgaveSomIkkeErFerdigstilt(
+        behandlingId: BehandlingId,
+        oppgavetyper: Set<Oppgavetype>,
+    ): OppgaveDomain? =
+        oppgaveRepository.findByBehandlingIdAndErFerdigstiltIsFalseAndTypeIn(
+            behandlingId,
+            oppgavetyper,
+        )
+
+    fun utledAnsvarligSaksbehandlerForOppgave(
+        behandlingId: BehandlingId,
+        behandleSakOppgave: OppgaveDomain?,
+    ): SaksbehandlerDto {
+        val rolle = utledSaksbehandlerRolle(behandlingId, behandleSakOppgave)
+
+        val tilordnetSaksbehandler =
+            if (rolle == SaksbehandlerRolle.OPPGAVE_FINNES_IKKE_SANNSYNLIGVIS_INNLOGGET_SAKSBEHANDLER) {
+                hentSaksbehandlerInfo(SikkerhetContext.hentSaksbehandler())
+            } else if (behandleSakOppgave?.tilordnetSaksbehandler == null) {
+                null
```

Comment:
Er vi alltid i OPPGAVE_FINNES_IKKE senario her? Kan vi i så fall legge til en kommentar om det? Vet jeg var med å skrive denne delen av koden, men må fortsatt bruke mye hjernekapasistet for å forstå 😅